### PR TITLE
add http scheme

### PR DESCRIPTION
**Pull request by mrmiywj**
#1536 
PTAL @siddontang @shenli

### CODE REVIEW

**Comment by mrmiywj** [comment]
PTAL @overvenus

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -807,10 +807,16 @@ fn main() {
         let addr = get_toml_string(&config,
                                    "server.addr",
                                    Some(DEFAULT_LISTENING_ADDR.to_owned()));
-        if let Err(e) = util::config::check_addr(&addr) {
+        let addr_ = if addr.starts_with("http://") {
+            let (_, a) = addr.split_at(7);
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: Why not `addr[7..].to_owned()`?

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -807,10 +807,16 @@ fn main() {
         let addr = get_toml_string(&config,
                                    "server.addr",
                                    Some(DEFAULT_LISTENING_ADDR.to_owned()));
-        if let Err(e) = util::config::check_addr(&addr) {
+        let addr_ = if addr.starts_with("http://") {
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: A postfix `_` is ugly.

Comment (siddontang) [inline src/bin/tikv-server.rs]: not `server.addr`, should handle this in `server.pd`

**Comment by mrmiywj** [comment]
PTAL @siddontang @BusyJay

File: src/bin/tikv-server.rs [hunk 3]
```diff
@@ -815,7 +815,10 @@ fn main() {
 
     let pd_endpoints = get_flag_string(&matches, "pd")
         .unwrap_or_else(|| get_toml_string(&config, "pd.endpoints", None));
-    for addr in pd_endpoints.split(',').map(|s| s.trim()).filter(|s| !s.is_empty()) {
+    for addr in pd_endpoints.split(',')
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: I think `fiter_map` should handle this properly.

**Comment by siddontang** [comment]
Any update? @mrmiywj

**Comment by mrmiywj** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay @overvenus

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**